internal/core/software_renderer.rs

```diff
@@ -768,11 +768,11 @@ fn render_window_frame_by_line(
             },
         );
 
-        if scene.current_line < to_draw_tr.origin.y_length() + to_draw_tr.size.height_length() {
+        if scene.current_line < dirty_region.origin.y_length() + dirty_region.size.height_length() {
             scene.next_line();
         }
     }
-    PhysicalRegion(dirty_region)
+    PhysicalRegion(to_draw_tr)
```

Comment:
I've return the transformed region, but note that the render function had a comment that explicitly stated that the region was the non-rotated region. Since the rotation is not public yet it's not a breaking change, but just a thing to note. I've also modified the doc to match the modifications